### PR TITLE
Use same region resolution for S3 and Kinesis

### LINKED ISSUE #1519: Use same logic for region resolution in S3 storage and Kinesis
Kinesis should align on S3 storage region resolution.

### PR DESCRIPTION
**Pull request by boraarslan**
Fixes #1519

### CODE REVIEW

File: quickwit-indexing/src/source/kinesis/kinesis_source.rs [hunk 1]
```diff
@@ -340,16 +341,23 @@ impl Source for KinesisSource {
...
+
+    // TODO: Change this to let_chain when 1.64 lands. (this is ugly)
+    if let Some(RegionOrEndpoint::Region(region)) = region_or_endpoint {
+        match region.parse() {
+            Ok(region_val) => return Ok(region_val),
+            Err(_) => {}
+        }
+    }
+
+    sniff_s3_region_and_cache() //< We fallback to S3 region if parsing from `region_or_endpoint`
```
Comment (boraarslan) [inline quickwit-indexing/src/source/kinesis/kinesis_source.rs:359]: If parsing from the given region fails, it tries to get the region from `sniff_s3_region`. Should I keep it this way and add some logs to notify the user or should I revert the change to fail completely if it fails to resolve?

File: quickwit-aws/src/region.rs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+use once_cell::sync::OnceCell;
+use rusoto_core::credential::ProfileProvider;
+use rusoto_core::Region;
+use tracing::{debug, error, info, instrument, warn};
+
+/// Default region to use, if none has been configured.
+const QUICKWIT_DEFAULT_REGION: Region = Region::UsEast1;
+
+#[instrument]
+fn sniff_s3_region() -> anyhow::Result<Region> {
```
Comment (guilload) [inline quickwit-aws/src/region.rs:33]: This is generic enough to be renamed `sniff_aws_region`

File: quickwit-aws/src/region.rs [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+/// One of this method runs a GET request on an EC2 API Endpoint.
+/// If this endpoint is inaccessible, this can block for a few seconds.
+///
+/// For this reason, this function needs to be called in a lazy manner.
+/// We cannot call it once when we create the
+/// `S3CompatibleObjectStorageFactory` for instance, as it would
+/// impact the start of quickwit in a context where S3 is not used.
+///
+/// This function caches its results.
+pub fn sniff_s3_region_and_cache() -> anyhow::Result<Region> {
```
Comment (guilload) [inline quickwit-aws/src/region.rs:66]: `sniff_aws_region_and_cache`

**Review by guilload — APPROVED** [review]
LGTM

**Pull request merged**